### DISCUSSION TITLE
Visualize GPS heading

### DISCUSSION THREAD

**Discussion by jhurliman** [foxglove/community#668] [Q&A]
It would be a useful feature addition to the Map panel if the `foxglove.LocationFix` message had a heading field (in radians counter-clockwise from north) and the current GPS point on the Map panel showed an oriented arrow similar to Google Maps / Apple Maps. It would probably need to be a toggle in the settings since there's no way to distinguish `heading=0` as direct north or uninitialized, unless there was an additional bool or enum field specifying if heading is provided.

**Comment by woudie-swap** [comment]
out of curiosity, has there been any progress on this?

**Reply by defunctzombie** [reply]
Nope

**Comment by ndmmdn** [comment]
I agree this feature would be quite useful, but seeing as per[ REP 103 ](https://www.ros.org/reps/rep-0103.html) the standard heading representation is in an ENU frame I would say heading=0 should point east (or otherwise make this somehow configurable)

**Reply by banisadr** [reply]
Thanks for pointing that out, I've added a comment to our internal tracking ticket.

**Comment by indirectlylit** [comment]
Hi folks - we're currently working on this.

In addition to visualizing heading angle, we're looking at ways to visualize heading angle uncertainty. Would folks on this thread find that useful? It could be visualized as a wedge:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/1a29b54c-26c9-4291-9fbe-9937f71763a6" />

A couple options for representation:

- a single `heading_variance` in rad². The wedge's half-angle is the standard deviation of the heading estimate (the square root of the variance), representing approximately 68% confidence. For example, a variance of 0.01 rad² would show a wedge with a half-angle of 0.1 rad (~6°)
- a left and right angle bounds, so that the wedge can be asymmetric relative to the `heading` angle

Another question is whether folks would expect only to specify the heading in the earth's tangent plane, or there is an expectation of being able to specify it in 3-dimensions? (A `LocationFix` technically has altitude, and its covariance matrix is 3D)

Curious to hear thoughts.

**Reply by woudie-swap** [reply]
Great to hear this is getting worked on now!

Have the heading uncertainty would be handy for sure and I would expect a 2d presentation on the tangent plane (at least for now).
